pkg/snapshot/copier/copier.go

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package copier
+
+import (
+	"fmt"
+
+	"github.com/gardener/etcd-backup-restore/pkg/miscellaneous"
+	"github.com/gardener/etcd-backup-restore/pkg/snapstore"
+	brtypes "github.com/gardener/etcd-backup-restore/pkg/types"
+
+	"github.com/sirupsen/logrus"
+)
+
+// GetSourceAndDestinationStores returns the source and destination stores for the given copier and store configs.
+func GetSourceAndDestinationStores(sourceSnapStoreConfig *brtypes.SnapstoreConfig, destSnapStoreConfig *brtypes.SnapstoreConfig) (brtypes.SnapStore, brtypes.SnapStore, error) {
+	sourceSnapStore, err := snapstore.GetSnapstore(sourceSnapStoreConfig)
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not get source snapstore: %v", err)
+	}
+
+	destSnapStore, err := snapstore.GetSnapstore(destSnapStoreConfig)
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not get destination snapstore: %v", err)
+	}
+
+	return sourceSnapStore, destSnapStore, nil
+}
+
+// Copier can be used to copy backups from a source to a destination store.
+type Copier struct {
+	logger          *logrus.Entry
+	sourceSnapStore brtypes.SnapStore
+	destSnapStore   brtypes.SnapStore
+}
+
+// NewCopier creates a new copier.
+func NewCopier(sourceSnapStore brtypes.SnapStore, destSnapStore brtypes.SnapStore, logger *logrus.Entry) *Copier {
+	return &Copier{
+		logger:          logger.WithField("actor", "copier"),
+		sourceSnapStore: sourceSnapStore,
+		destSnapStore:   destSnapStore,
+	}
+}
+
+// Run executes the copy command.
+func (c *Copier) Run() error {
+	return c.CopyBackups()
+}
+
+// CopyBackups copies all backups from the source store to the destination store.
+func (c *Copier) CopyBackups() error {
+	// Get source backups
+	c.logger.Info("Getting source backups...")
+	sourceBackups, err := miscellaneous.GetAllBackups(c.sourceSnapStore)
```

Comment:
Small nit, but we should probably unify the names between `sourceBackups` and the destination `snapshots` on line 80